src/map/lua/luautils.cpp

```diff
@@ -5724,4 +5732,47 @@ namespace luautils
 
         return CLuaBaseEntity(PEntity);
     }
+
+    // Fishing Contest utilities
+    auto GetFishingContest() -> sol::table
+    {
+        sol::table table = lua.create_table();
+
+        table["status"]     = fishingcontest::GetContestStatus();
+        table["criteria"]   = fishingcontest::GetContestCriteria();
+        table["measure"]    = fishingcontest::GetContestMeasure();
+        table["fishid"]     = fishingcontest::GetContestFish();
+        table["starttime"]  = fishingcontest::GetContestStartTime();
+        table["changetime"] = fishingcontest::GetContestChangeTime();
+
+        return table;
+    }
+
+    void InitNewFishingContest()
+    {
+        // This function will destroy the current contest and start a new one
+        fishingcontest::InitNewContest();
+    }
+
+    void SetContestParameters(uint16 fishId, uint8 measure, uint8 criteria)
+    {
+        // Note: Validation of fish item id should be done in lua, since the fish table is also in lua
+        fishingcontest::SetContestFish(fishId);
+        fishingcontest::SetContestMeasure((FISHING_CONTEST_MEASURE)std::min((uint8)measure, (uint8)CONTEST_MEASURE_SMALLEST));
+        fishingcontest::SetContestCriteria((FISHING_CONTEST_CRITERIA)std::min((uint8)criteria, (uint8)CONTEST_CRITERIA_BOTH));
```

Comment:
Don't use c-style casting, especially not if you have enums. Make your function take the enums as their args

Reply:
Not sure I understand this one -- the `std::min` function wont take the enums as they are, but spits out an `int`, hence the casting to the enum to pass into the function.

Reply:
Don't worry about it for now, this'll need extra plumbing in core for the safety I eventually want. Just make sure you replace your c-style casts with static_casts etc.

(the plumbing I'm looking at: https://github.com/Neargye/magic_enum)